alchemist/alchemist-implementationbase/src/main/kotlin/it/unibo/alchemist/model/interfaces/geometry/euclidean/twod/Segment2D.kt

```diff
@@ -14,84 +16,84 @@ data class Segment2D<P : Vector2D<P>>(val first: P, val second: P) {
     /**
      * Checks whether the segment is aligned to the x axis.
      */
-    val xAxisAligned: Boolean get() = MathUtils.fuzzyEquals(first.y, second.y)
+    val xAxisAligned: Boolean by lazy { MathUtils.fuzzyEquals(first.y, second.y) }
 
     /**
      * Checks whether the segment is aligned to the y axis.
      */
-    val yAxisAligned: Boolean get() = MathUtils.fuzzyEquals(first.x, second.x)
+    val yAxisAligned: Boolean by lazy { MathUtils.fuzzyEquals(first.x, second.x) }
 
     /**
      * Checks whether the segment is axis-aligned.
      */
-    val isAxisAligned: Boolean get() = xAxisAligned || yAxisAligned
+    val isAxisAligned: Boolean by lazy { xAxisAligned || yAxisAligned }
 
     /**
-     * Mutates an edge to a vector. In particular, the vector representing the
-     * movement from the first point to the second point of the edge.
+     * @returns the vector representing the movement from [first] to [second].
      */
     fun toVector() = second - first
 
     /**
-     * Computes the slope of the line passing through a couple of points.
-     * If the points coincide NaN is the result.
+     * Computes the slope of the segment. If the two points coincide (i.e. the segment
+     * [isDegenerate]), [Double.NaN] is the result.
      */
-    val slope: Double get() = toVector().run { y / x }
+    val slope: Double by lazy { toVector().run { y / x } }
 
     /**
-     * An edge is degenerate if its points coincide (and its length is zero).
+     * Computes the intercept of the line passing through [first] and [second].
      */
-    val degenerate: Boolean get() = first == second
+    val intercept: Double by lazy { first.y - slope * first.x }
 
     /**
-     * Checks whether the segment (represented by a pair of positions)
-     * contains the given point.
+     * Checks if its points coincide (and its length is zero).
      */
-    fun contains(other: P) =
-        areCollinear(first, second, other) &&
-        other.x.liesBetween(first.x, second.x) &&
-        other.y.liesBetween(first.y, second.y)
+    val isDegenerate: Boolean by lazy { first == second }
```

Comment:
This is actually way worse than computing every time. The cost is higher than a lazy delegate: is there a reason other than performance for the change?